src/redux/books/bookSlice.js

```diff
@@ -0,0 +1,108 @@
+/* eslint-disable no-param-reassign */
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+import API_URL from '../../api';
+
+export const postBook = createAsyncThunk(
+  'books/postBook',
+  async (bookData, thunkAPI) => {
+    try {
+      const res = await axios.post(`${API_URL}/books`, bookData);
+
+      return res.data;
+    } catch (error) {
+      return thunkAPI.rejectWithValue(
+        error?.data?.message || 'Something went wrong!',
+      );
+    }
+  },
+);
+
+export const getBooks = createAsyncThunk(
+  'books/getBooks',
+  async (_, thunkAPI) => {
+    try {
+      const res = await axios(`${API_URL}/books`);
+      return res.data;
+    } catch (error) {
+      return thunkAPI.rejectWithValue(
+        error?.data?.message || 'Something went wrong!',
+      );
+    }
+  },
+);
+
+export const deleteBook = createAsyncThunk(
+  'books/deleteBook',
+  async (id, thunkAPI) => {
+    try {
+      const res = await axios.delete(`${API_URL}/books/${id}`);
+
+      return res.data;
+    } catch (error) {
+      return thunkAPI.rejectWithValue(
+        error?.data?.message || 'Something went wrong!',
+      );
+    }
+  },
+);
+
+const initialState = {
+  isLoading: false,
+  books: [],
+};
+
+const booksSlice = createSlice({
+  name: 'books',
+  initialState,
+  reducers: {
+    addBook: (state, actions) => {
+      const bookData = actions.payload;
+      state.books.push(bookData);
+    },
+
+    removeBook: (state, actions) => {
+      const idOfBookToRemove = actions.payload;
+      state.books = state.books.filter(
+        (book) => book.item_id !== idOfBookToRemove,
+      );
+    },
+  },
+  extraReducers: (builder) => {
+    builder
+      .addCase(postBook.pending, (state) => {
+        state.isLoading = true;
+      })
+      .addCase(postBook.fulfilled, (state) => {
+        state.isLoading = false;
+      })
+      .addCase(postBook.rejected, (state) => {
+        state.isLoading = false;
+      });
+
+    builder
+      .addCase(getBooks.pending, (state) => {
+        state.isLoading = true;
+      })
+      .addCase(getBooks.fulfilled, (state, action) => {
+        state.isLoading = false;
+        const resObject = action.payload;
+
+        const newBooksArr = [];
+        // eslint-disable-next-line no-restricted-syntax, guard-for-in
```

Comment:
**You did great job! Almost there :)**

- [ ] However we have some placed to fix up. For starting, kindly remove all `// eslint-disable...` from your project. 

> 🔧 Sometimes we may encounter challenges when using ESLint. Unexpected errors may arise which can be difficult to resolve quickly. If you come across an issue that is functioning properly but is still displaying an error message when uploaded to GitHub, I recommend disabling it in the .eslintrc.json file. This can help to prevent any confusion or delays in the development process. Let me know if you need any assistance.